chart/values.yaml

```diff
@@ -35,12 +35,21 @@ revisionHistoryLimit: ~
 uid: 50000
 gid: 0
 
-# Default security context for airflow
+# Default security context for airflow (deprecated, use below in the future)
 securityContext: {}
 #  runAsUser: 50000
 #  fsGroup: 0
 #  runAsGroup: 0
 
+# Detailed default security context for airflow deployments
+securityContexts:
+  pod: {}
+  containers:
+    allowPrivilegeEscalation: false
+    capabilities:
+      drop:
+        - ALL
```

Comment:
This is the default, yeah? Not sure we need to expand it here.

Reply:
I removed it from here and put it in the helper:
{{/*
Set the default value for container securityContext
If no value is passed for securityContext.container or <node>.securityContext.container, defaults to deny privileges escallation and dropping all POSIX capabilities.

    +------------------------+      +-----------------+      +-------------------------+
    | <node>.securityContext.container |  ->  | securityContext.container |  ->  | allowPrivilegesEscalation: false, capabilities.drop: [ALL]|
    +------------------------+      +-----------------+      +-------------------------+


The template can be called like so:
   include "airflowSecurityContextContainer" (list . .Values.webserver)

Where `.` is the global variables scope and `.Values.webserver` the local variables scope for the webserver template.
*/}}
{{- define "airflowSecurityContextContainer" -}}
  {{- $ := index . 0 -}}
  {{- with index . 1 }}
    {{- if .securityContext.container -}}
{{ toYaml .securityContext.container | print }}
    {{- else if $.Values.securityContext.container -}}
{{ toYaml $.Values.securityContext.container | print }}
    {{- else -}}
allowPrivilegeEscalation: false
capabilities:
  drop:
    - ALL
    {{- end -}}
  {{- end -}}
{{- end -}}

Reply:
Sorry, should have been more clear. Isn't that the default kubernetes sets if you provide nothing? If so, I'd rather we not set it at all.

Reply:
I think you have to set them yourself

Reply:
Sorry for the delay, coming back to this now.

I had this backwards - this removes rights. While I agree this is a good sane default, we can't make this change in a minor release without breaking backward compatibility.  Given that, I think for now we should keep the default as nothing.